core/src/banking_stage.rs

```diff
@@ -1190,6 +1216,71 @@ impl BankingStage {
             .collect()
     }
 
+    fn generate_vote_packet_indexes(vers: &PinnedVec<Packet>) -> Vec<usize> {
+        vers.iter()
+            .enumerate()
+            .filter_map(|(index, ver)| {
+                if !ver.meta.discard {
+                    let tx: Transaction = limited_deserialize(&ver.data[0..ver.meta.size]).ok()?;
+                    if is_simple_vote_transaction(&tx) {
+                        Some(index)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn process_packets_vote(
```

Comment:
Maybe a basic unit test to test it correctly 
1) parses out vote transactions from a group
2) Runs the proper sanity checks

Reply:
unit tested